csv/write_read_test.go

```diff
@@ -15,61 +20,78 @@ func TestWriteRead(t *testing.T) {
 		options     []Options
 		outputCount int
 	}{
-		{name: "default", outputCount: 1},
-		{name: "with_headers", options: []Options{WithHeader()}, outputCount: 1},
-		{name: "with_delimiter", options: []Options{WithDelimiter('\t')}, outputCount: 1},
-		{name: "with_delimter_headers", options: []Options{WithDelimiter('\t'), WithHeader()}, outputCount: 1},
+		{name: "default", outputCount: 2},
+		{name: "with_headers", options: []Options{WithHeader()}, outputCount: 2},
+		{name: "with_delimiter", options: []Options{WithDelimiter('\t')}, outputCount: 2},
+		{name: "with_delimiter_headers", options: []Options{WithDelimiter('\t'), WithHeader()}, outputCount: 2},
 	}
 
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
-			var b bytes.Buffer
 			table := testdata.TestTable("test")
-			cqtypes := testdata.GenTestData(table)
-			if err := cqtypes[0].Set("test-source"); err != nil {
+			arrowSchema := table.ToArrowSchema()
+			sourceName := "test-source"
+			syncTime := time.Date(2021, 1, 1, 0, 0, 0, 0, time.UTC)
+			mem := memory.NewCheckedAllocator(memory.NewGoAllocator())
+			defer mem.AssertSize(t, 0)
+			opts := testdata.GenTestDataOptions{
+				SourceName: sourceName,
+				SyncTime:   syncTime,
+				MaxRows:    2,
+				StableUUID: uuid.MustParse("00000000-0000-0000-0000-000000000001"),
+				StableTime: time.Date(2021, 1, 2, 0, 0, 0, 0, time.UTC),
+			}
+			records := testdata.GenTestData(mem, arrowSchema, opts)
+			defer func() {
+				for _, r := range records {
+					r.Release()
+				}
+			}()
+			cl, err := NewClient(tc.options...)
+			if err != nil {
 				t.Fatal(err)
 			}
+
+			var b bytes.Buffer
 			writer := bufio.NewWriter(&b)
 			reader := bufio.NewReader(&b)
-			transformer := &schema.DefaultTransformer{}
-			transformedValues := schema.TransformWithTransformer(transformer, cqtypes)
-			client, err := NewClient(tc.options...)
-			if err != nil {
+
+			if err := cl.WriteTableBatch(writer, arrowSchema, records); err != nil {
 				t.Fatal(err)
 			}
+			writer.Flush()
 
-			if err := client.WriteTableBatch(writer, table, [][]any{transformedValues}); err != nil {
+			rawBytes, err := io.ReadAll(reader)
+			if err != nil {
 				t.Fatal(err)
 			}
-			writer.Flush()
+			snap := cupaloy.New(cupaloy.SnapshotFileExtension(".csv"))
```

Comment:
```suggestion
			snap := cupaloy.New(cupaloy.SnapshotFileExtension(".csv"), cupaloy.SnapshotSubdirectory("testdata"))
```
Why not use `testdata/` for the test data? `.snapshots` is a bit vague

Reply:
We use `.snapshots` in the plugin-sdk as well, but yeah I don't really have a strong preference here

Reply:
Updated to use `testdata`